components/brave_wallet/browser/solana_response_parser.cc

```diff
@@ -185,6 +185,55 @@ bool ParseGetSignatureStatuses(
   return true;
 }
 
+bool ParseGetAccountInfo(const std::string& json,
+                         absl::optional<SolanaAccountInfo>* account_info_out) {
+  DCHECK(account_info_out);
+
+  base::Value result;
+  if (!ParseResult(json, &result) || !result.is_dict())
+    return false;
+
+  const base::Value* value = result.FindKey("value");
+  if (!value || (!value->is_none() && !value->is_dict()))
+    return false;
+
+  if (value->is_none()) {
+    *account_info_out = absl::nullopt;
+    return true;
+  }
+
+  SolanaAccountInfo account_info;
+  if (!GetUint64FromDictValue(*value, "lamports", false,
+                              &account_info.lamports))
+    return false;
+
+  auto* owner = value->FindStringKey("owner");
+  if (!owner)
+    return false;
+  account_info.owner = *owner;
+
+  auto* data = value->FindListKey("data");
+  if (!data)
+    return false;
+  const auto& data_list = data->GetList();
+  if (data_list.size() != 2u || !data_list[0].is_string() ||
+      !data_list[1].is_string() || data_list[1].GetString() != "base64")
+    return false;
+  account_info.data = data_list[0].GetString();
```

Comment:
I think we need to check if the data is really base64 encode here by `Base64Decode`
and add a test case for that

Reply:
Addressed in https://github.com/brave/brave-core/pull/12729/commits/1ce4e199f80ca096af5836e593bb407bc1817b74